docs/deprecation.md

```diff
@@ -0,0 +1,84 @@
+# External DNS Deprecation Policy
+
+This document defines the Deprecation Policy for External Secrets Operator Components.
```

Comment:
```suggestion
This document defines the Deprecation Policy for External DNS.
```

Reply:
Now I know where does this document comes from :see_no_evil:

Reply:
Yep. There is no point to introduce things. ;-)